miner/worker.go

```diff
@@ -775,11 +785,237 @@ func (w *worker) updateSnapshot() {
 func (w *worker) commitTransaction(tx *types.Transaction, coinbase common.Address) ([]*types.Log, error) {
 	snap := w.current.state.Snapshot()
 
+	// reset StructLogger in case of OOM
+	tracer := w.chain.GetVMConfig().Tracer.(*vm.StructLogger)
+	tracer.Reset()
+
+	zkTrieTracers := make(map[string]state.ZktrieProofTracer)
+
+	from, _ := types.Sender(w.current.signer, tx)
+	to := tx.To()
+
+	sender := &types.AccountWrapper{
+		Address:          from,
+		Nonce:            w.current.state.GetNonce(from),
+		Balance:          (*hexutil.Big)(w.current.state.GetBalance(from)),
+		KeccakCodeHash:   w.current.state.GetKeccakCodeHash(from),
+		PoseidonCodeHash: w.current.state.GetPoseidonCodeHash(from),
+		CodeSize:         w.current.state.GetCodeSize(from),
+	}
+	var receiver *types.AccountWrapper
+	if to != nil {
+		receiver = &types.AccountWrapper{
+			Address:          *to,
+			Nonce:            w.current.state.GetNonce(*to),
+			Balance:          (*hexutil.Big)(w.current.state.GetBalance(*to)),
+			KeccakCodeHash:   w.current.state.GetKeccakCodeHash(*to),
+			PoseidonCodeHash: w.current.state.GetPoseidonCodeHash(*to),
+			CodeSize:         w.current.state.GetCodeSize(*to),
+		}
+	}
+
+	// currently still need to add storageTrace into the traces, but only with the coinbase proof
+	storageTrace := &types.StorageTrace{Proofs: make(map[string][]hexutil.Bytes)}
+	if _, exist := storageTrace.Proofs[coinbase.String()]; !exist {
+		proof, err := w.current.state.GetProof(coinbase)
+		if err != nil {
+			log.Error("Proof for coinbase not available", "coinbase", coinbase, "error", err)
+			// but we still mark the proofs map with nil array
+		}
+		wrappedProof := make([]hexutil.Bytes, len(proof))
+		for i, bt := range proof {
+			wrappedProof[i] = bt
+		}
+		storageTrace.Proofs[coinbase.String()] = wrappedProof
+	}
+
+	txStorageTrace := &types.StorageTrace{
+		RootBefore:    w.current.state.GetRootHash(),
+		Proofs:        make(map[string][]hexutil.Bytes),
+		StorageProofs: make(map[string]map[string][]hexutil.Bytes),
+	}
+
 	receipt, err := core.ApplyTransaction(w.chainConfig, w.chain, &coinbase, w.current.gasPool, w.current.state, w.current.header, tx, &w.current.header.GasUsed, *w.chain.GetVMConfig())
 	if err != nil {
 		w.current.state.RevertToSnapshot(snap)
 		return nil, err
 	}
+
+	txStorageTrace.RootAfter = w.current.state.GetRootHash()
```

Comment:
i think the GetRootHash() will return the same root with RootBefore,  but it is ok since in capacity checker, i will not use this currently.   Besides,   calculating the "real state root after tx" will be expensive.

Reply:
fixed in https://github.com/scroll-tech/go-ethereum/commit/4623d904f412c517d4439d6b405bac563e538220